kolibri/auth/test/test_permissions_classes.py

```diff
@@ -46,23 +47,26 @@ def test_user_cannot_delete(self):
 class TestBooleanOperationsOnPermissionClassesTestCase(TestCase):
 
     def setUp(self):
+        self.facility = Facility.objects.create()
         self.obj = object()
-        self.user = FacilityUser.objects.create(username='dummyuser')
+        self.user = FacilityUser.objects.create(username='dummyuser', facility=self.facility)
         self.queryset = FacilityUser.objects.all()
 
-    def assertAllowAll(self, perms):
+    def assertAllowAll(self, perms, test_filtering=True):
         self.assertTrue(perms.user_can_create_object(self.user, self.obj))
         self.assertTrue(perms.user_can_read_object(self.user, self.obj))
         self.assertTrue(perms.user_can_update_object(self.user, self.obj))
         self.assertTrue(perms.user_can_delete_object(self.user, self.obj))
-        self.assertSetEqual(self.queryset, perms.readable_by_user_filter(self.user, self.queryset))
+        if test_filtering:
+            self.assertSetEqual(set(self.queryset), set(perms.readable_by_user_filter(self.user, self.queryset)))
```

Comment:
Cool, thanks -- wasn't familiar with `assertQuerysetEqual`. I tried it out, and it's a bit weird in how it compares things; the second argument is by default a list of `repr` strings of the model instances. I would have to specify a different transform for it to work, which starts to get kludgy:
`self.assertQuerysetEqual(self.queryset, list(perms.readable_by_user_filter(self.user, self.queryset)), transform=lambda x: x, ordered=False)`

Since the two querysets are for the same Model, and pk's are unique within a Model, it should be safe to keep it as is in this case, but I can switch it to the line shown above if preferred.